projects/laji/src/app/shared-modules/tree-select/tree-select.component.ts

```diff
@@ -47,15 +49,24 @@ export class TreeSelectComponent {
 
   lang: string;
   modalRef: BsModalRef;
+  options: SelectedOption[];
```

Comment:
I'm a bit hesitant about making the `options` a non-observable. Whether to use observables with `async` pipe or just updating the value and fiddle with manual change detection is up to debate, but usually the former is cleaner IMO. 

Usual up-sides of using observables instead of vanilla objects:
* declarative style: control of logic is stated in one place and not scattered around
* no need for manual change detection checks
* race conditions are easier to handle

In this case I'm not that strongly against what you did here, because for example there shouldn't be any race conditions since in practice the initialization from `this.options$` is always ran before the component is rendered and the user could trigger any events that would update the value before the initialization. In general I feel that using observables is the best way because it's more maintainable if the logic gets more complex in the future.

I think you don't need the change detection mark at `ngOnInit` because the component uses the `OnPush` strategy and `this.options` is updated in an immutable way? Not sure, maybe the update being done inside a subscription makes Angular miss the update?

Reply:
Will look more into refactoring this some other time.